### PR TITLE
Remove driver level PIO initialization

### PR DESCRIPTION
**Pull request by uturuncoglu**
This PR aims to remove the modifications that was done previously to bring PIO initialization to NEMS driver. Since both CMEPS (https://github.com/hafs-community/CMEPS/pull/1) and CDEPS (https://github.com/hafs-community/CDEPS/pull/2) are modified to have component level PIO initialization, there is no need to keep those modifications in the NEMS level.

Tests:
- Full RTs were run without any problem and passed.

### CODE REVIEW

**Review by BinLiu-NOAA — APPROVED** [review]
Looks good to me.

**Comment by BinLiu-NOAA** [comment]
@uturuncoglu In order to not break the support/HAFS branch at the model level, please hold this PR so that it goes back into the support/HAFS branch together with the following PRs:
hafs-community/ufs-weather-model#8
hafs-community/CMEPS#1
hafs-community/CDEPS#2

**Comment by uturuncoglu** [comment]
@BinLiu-NOAA sure. I am waiting for your approval.

File: src/MAIN_NEMS.F90
Comment (danrosen25) [inline src/MAIN_NEMS.F90:206]: Please remove line 121 if no longer needed. INTEGER :: COMM_WORLD

File: src/module_EARTH_GRID_COMP.F90
Comment (danrosen25) [inline src/module_EARTH_GRID_COMP.F90:3439]: VM is no longer needed. See lines
- [ ] 3407
- [ ] 3426
- [ ] 4576

**Review by danrosen25 — CHANGES_REQUESTED** [review]
I suggest cleaning up a few hanging variables before approving.

**Comment by uturuncoglu** [comment]
@BinLiu-NOAA @danrosen25 i am closing this PR because this branch is outdated and I need to work again. I'll create another branch for CDEPS sync and make PR again. Then we could clean the unused branches.

**Pull request closed without merging**
